plugins/processors/enum/enum.go

```diff
@@ -0,0 +1,110 @@
+package enum
+
+import (
+	"strconv"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+var sampleConfig = `
+## NOTE This processor will map metric values to different values. It is aimed
+## to map enum values to numeric values.
+
+## Fields to be considered
+# [[processors.enum.fields]]
+#
+# Name of the field
+#   key = "name"
+#
+# Value Mapping Table
+#   [processors.enum.value_mappings]
+#     value1 = 1
+#     value2 = 2
+#
+## Alternatively the mapping table can be given in inline notation
+#   value_mappings = {value1 = 1, value2 = 2}
+`
+
+type EnumMapper struct {
+	Fields []Mapping
+}
+
+type Mapping struct {
+	Key           string
+	ValueMappings map[string]interface{}
+}
+
+func (mapper *EnumMapper) SampleConfig() string {
+	return sampleConfig
+}
+
+func (mapper *EnumMapper) Description() string {
+	return "Map enum values according to given table."
+}
+
+func (mapper *EnumMapper) Apply(in ...telegraf.Metric) []telegraf.Metric {
+	out := make([]telegraf.Metric, 0, len(in))
```

Comment:
We should try to make this as efficient as possible since it will be ran in over all metrics, perhaps we can modify `in` directly?